packages/react/src/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license EUPL-1.2+
+ * Copyright (c) 2023 Gemeente Amsterdam
+ */
+
+import { CloseIcon } from '@amsterdam/design-system-react-icons'
+import clsx from 'clsx'
+import { DialogHTMLAttributes, ForwardedRef, forwardRef, PropsWithChildren, ReactNode } from 'react'
+import { Icon } from '../Icon'
+import { VisuallyHidden } from '../VisuallyHidden'
+
+export interface DialogProps extends PropsWithChildren<DialogHTMLAttributes<HTMLDialogElement>> {
+  actions?: ReactNode
+}
+
+const DialogClose = forwardRef(({ ...restProps }, ref: ForwardedRef<HTMLButtonElement>) => (
+  <button role="close" aria-label="close" ref={ref} className="amsterdam-dialog__close" formNoValidate {...restProps}>
```

Comment:
I don't think making up your own `role` is a good idea, screenreaders ignore it. 

Having both `aria-label` and `VisuallyHidden` seems a bit much, don't we just want one or the other? `VisuallyHidden` has better support, I think